ibc-core/ics02-client/types/src/msgs/upgrade_client.rs

```diff
@@ -59,26 +60,34 @@ impl TryFrom<RawMsgUpgradeClient> for MsgUpgradeClient {
     fn try_from(proto_msg: RawMsgUpgradeClient) -> Result<Self, Self::Error> {
         let raw_client_state = proto_msg
             .client_state
-            .ok_or(ClientError::MissingRawClientState)?;
+            .ok_or(DecodingError::MissingRawData {
+                description: "client state not set".to_string(),
+            })?;
 
-        let raw_consensus_state = proto_msg
-            .consensus_state
-            .ok_or(ClientError::MissingRawConsensusState)?;
+        let raw_consensus_state =
+            proto_msg
+                .consensus_state
+                .ok_or(DecodingError::MissingRawData {
+                    description: "consensus state not set".to_string(),
+                })?;
 
         let c_bytes =
             CommitmentProofBytes::try_from(proto_msg.proof_upgrade_client).map_err(|_| {
-                UpgradeClientError::InvalidUpgradeClientProof(CommitmentError::EmptyMerkleProof)
+                UpgradeClientError::InvalidUpgradeClientStateProof(
+                    CommitmentError::InvalidMerkleProof,
+                )
             })?;
+
         let cs_bytes = CommitmentProofBytes::try_from(proto_msg.proof_upgrade_consensus_state)
             .map_err(|_| {
                 UpgradeClientError::InvalidUpgradeConsensusStateProof(
-                    CommitmentError::EmptyMerkleProof,
+                    CommitmentError::InvalidMerkleProof,
```

Comment:
Since the only reason a `CommitmentProofBytes` might fail is if it's empty, the previous `EmptyMerkleProof` has been fine.

Reply:
On a second thought, the `InvalidMerkleProof` is alright too.